templates/default/datadog.yaml.erb

```diff
@@ -136,6 +137,10 @@ if !http_proxy.nil?
   }
 end
 
+if !no_proxy.nil?
+  agent_config['proxy']['no_proxy'] = no_proxy
```

Comment:
`agent_config['proxy']` should be initialized to a hash here, but the logic is a bit brittle. Could you either check that the key is indeed present in `agent_config` or move this block inside the `!http_proxy.nil?` conditional above?